src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/InstantiateArgumentExceptionsCorrectlyTests.cs

```diff
@@ -368,6 +593,116 @@ End Sub
                End Class");
         }
 
+        [Fact]
+        public async Task ArgumentException_GenericParameterName_DoesNotWarn()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+                public class Class
+                {
+                    public void Test<TEnum>(TEnum first)
+                    {
+                        throw new System.ArgumentException(""first is incorrect"", nameof(TEnum));
+                    }
+                }");
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+                Public Class [MyClass]
+                    Public Sub Test(Of TEnum)(ByVal first As TEnum)
+                        Throw New System.ArgumentException(""first is incorrect"", NameOf(TEnum))
+                    End Sub
+                End Class");
+        }
+
+
+        [Theory]
+        [InlineData("public", "dotnet_code_quality.api_surface = private", false)]
```

Comment:
Please add a test where 2nd argument is empty, i.e. no end user configuration, and validate that diagnostic is reported if first argument is "public" or "protected", but not when it is "private" or "internal". Basically test that the default configuration is enabling analysis only for public API surface.

Reply:
As mentioned before our default visibility intended to be `All` so it would report diagnostic on private or internal, so added cases for empty confiuration that suceed for all (didn't added for internal as its different in VB)